staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -333,6 +345,254 @@ func TestApfCancelWaitRequest(t *testing.T) {
+		}
+
+		var executed bool
+		// we will raise a panic for the first request.
+		firstRequestPathPanic := "/request/panic"
+		requestHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			executed = true
+			expectMatchingAPFHeaders(t, w, fsName, plName)
+
+			if r.URL.Path == firstRequestPathPanic {
+				panic(fmt.Errorf("request handler panic'd - %#v", r.RequestURI))
+			}
+		})
+		handler := newHandlerChain(t, requestHandler, controller, userName, requestTimeout)
+
+		server, requestGetter := newHTTP2ServerWithClient(handler)
+		defer server.Close()
+
+		var err error
+		_, err = requestGetter(firstRequestPathPanic)
+		if !executed {
+			t.Errorf("expected inner handler to be executed for request: %s", firstRequestPathPanic)
+		}
+		expectResetStreamError(t, err)
+
+		executed = false
+		// the second request should be served successfully.
+		secondRequestPathShouldWork := "/request/should-work"
+		response, err := requestGetter(secondRequestPathShouldWork)
+		if !executed {
+			t.Errorf("expected inner handler to be executed for request: %s", secondRequestPathShouldWork)
+		}
+		if err != nil {
+			t.Errorf("expected request: %s to succeed, but got error: %#v", secondRequestPathShouldWork, err)
+		}
+		if response.StatusCode != http.StatusOK {
+			t.Errorf("expected HTTP status code: %d for request: %s, but got: %#v", http.StatusOK, secondRequestPathShouldWork, response)
+		}
+
+		close(stopCh)
+		t.Log("waiting for the controller to shutdown gracefully")
+		<-stoppedCh
+
+		if controllerErr != nil {
+			t.Errorf("expected a nil error from controller, but got: %#v", controllerErr)
+		}
+	})
+}
+
+// returns the server concurrency that maps 1 to 1 with the concurrency shares.
+// it takes into account the concurrency share of the catch-all priority level and ensures that
+// the specified desired concurrency share = concurrency limit
+func getServerConcurrency(desiredShare int) int {
+	catchAllShare := fcboot.MandatoryPriorityLevelConfigurationCatchAll.Spec.Limited.AssuredConcurrencyShares
+	return int(catchAllShare) + desiredShare
+}
+
+// returns a started http2 server, with a client function to send request to the server.
+func newHTTP2ServerWithClient(handler http.Handler) (*httptest.Server, func(path string) (*http.Response, error)) {
+	server := httptest.NewUnstartedServer(handler)
+	server.EnableHTTP2 = true
+	server.StartTLS()
+
+	return server, func(path string) (*http.Response, error) {
+		return server.Client().Get(server.URL + path)
+	}
+}
+
+// verifies that the expected flow schema and priority level UIDs are attached to the header.
+func expectMatchingAPFHeaders(t *testing.T, w http.ResponseWriter, expectedFS, expectedPL string) {
+	if w == nil {
+		t.Fatal("expected a non nil HTTP response")
+	}
+
+	key := flowcontrol.ResponseHeaderMatchedFlowSchemaUID
+	if value := w.Header().Get(key); expectedFS != value {
+		t.Fatalf("expected an HTTP header: %s=%s, but got %s=%s", key, expectedFS, key, value)
```

Comment:
I am not a fan of repeating information.  I _am_ a fan of quoting unexpected strings.  How about something like the following?

> t.Fatalf("Expected HTTP header %s to have value %q but actually got %q", key, expectedFS, value)